services/brig/src/Brig/App.hs

```diff
@@ -549,45 +586,46 @@ instance Cas.MonadClient HttpClientIO where
 
 wrapHttpClient ::
   HttpClientIO a ->
-  AppT r IO a
+  AppT r a
 wrapHttpClient = wrapHttp
 
-wrapHttpClientE :: ExceptT e HttpClientIO a -> ExceptT e (AppT r IO) a
+wrapHttpClientE :: ExceptT e HttpClientIO a -> ExceptT e (AppT r) a
 wrapHttpClientE = mapExceptT wrapHttpClient
 
 instance MonadIO m => MonadIndexIO (ReaderT Env m) where
   liftIndexIO m = view indexEnv >>= \e -> runIndexIO e m
 
-instance MonadIndexIO (AppIO r) where
-  liftIndexIO m = AppT $ liftIndexIO m
+instance MonadIndexIO (AppT r) where
+  liftIndexIO m = do
+    AppT $ mapReaderT (embedToFinal @IO) $ liftIndexIO m
 
-instance (MonadIndexIO (AppT r m), Monad m) => MonadIndexIO (ExceptT err (AppT r m)) where
+instance MonadIndexIO (AppT r) => MonadIndexIO (ExceptT err (AppT r)) where
   liftIndexIO m = view indexEnv >>= \e -> runIndexIO e m
 
-instance Monad m => HasRequestId (AppT r m) where
+instance Monad m => HasRequestId (AppT r) where
   getRequestId = view requestId
 
-runAppT :: Env -> AppT r m a -> m a
-runAppT e (AppT ma) = runReaderT ma e
+runAppT :: Env -> AppT '[Final IO] a -> IO a
+runAppT e (AppT ma) = runFinal $ runReaderT ma e
 
-runAppResourceT :: ResourceT (AppIO r) a -> (AppIO r) a
+runAppResourceT :: ResourceT (AppT '[Final IO]) a -> (AppT '[Final IO]) a
 runAppResourceT ma = do
   e <- ask
   liftIO . runResourceT $ transResourceT (runAppT e) ma
 
-forkAppIO ::
+forkAppT ::
   (MonadIO m, MonadUnliftIO m, MonadReader Env m) =>
   Maybe UserId ->
   m a ->
   m ()
-forkAppIO u ma = do
+forkAppT u ma = do
```

Comment:
Do we still use this function?

Reply:
Thanks! I changed it to `fork` in commit b28740f42.